test/http-api/over-ipfs-api/bootstrap.js

```diff
@@ -28,6 +28,30 @@ module.exports = (ctl) => {
         })
       })
 
+      it('prevents duplicate inserts of bootstrap peers', () => {
+        return ctl
+          .bootstrap
+          .rm(null, { all: true })
+          .then((res) => {
+            expect(res.Peers.length).to.equal(0)
+            return ctl.bootstrap.add(validIp4)
+          })
+          .then(res => {
+            expect(res).to.be.eql({ Peers: [validIp4] })
+            return ctl.bootstrap.add(validIp4)
+          })
+          .then((res) => {
+            expect(res).to.be.eql({ Peers: [validIp4] })
+            return ctl.bootstrap.list()
+          })
+          .then((res) => {
+            expect(res).to.exist()
+            const insertPosition = res.Peers.indexOf(validIp4)
+            expect(insertPosition).to.not.equal(-1)
+            expect(res.Peers.length).to.equal(1)
+          })
+      })
```

Comment:
Please write tests for `core` too. The ideal would be getting `interface-ipfs-core` tests for bootstrap - https://github.com/ipfs/interface-ipfs-core/tree/master/src so that we match expectations on js-ipfs-api and js-ipfs.